tutorials/openmm-torch-nnpops.ipynb

```diff
@@ -0,0 +1,440 @@
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 1,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "EcB2lc1s3ZXG",
+        "outputId": "1ce0bad8-a050-48d5-c18a-9e49fbeb11c2"
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "⏬ Downloading https://repo.anaconda.com/miniconda/Miniconda3-py37_4.9.2-Linux-x86_64.sh...\n",
+            "📦 Installing...\n",
+            "📌 Adjusting configuration...\n",
+            "🩹 Patching environment...\n",
+            "⏲ Done in 0:00:21\n",
+            "🔁 Restarting kernel...\n"
+          ]
+        }
+      ],
+      "source": [
+        "!pip install -q condacolab\n",
+        "import condacolab\n",
+        "condacolab.install_miniconda()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "tWlPu4KF5DFZ"
+      },
+      "source": [
+        "## Install software\n",
+        "\n",
+        "The [conda-forge](https://conda-forge.org/) channel is used for software.\n",
+        "\n",
+        "⚠️ The installation might take up to 10 min!"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 2,
+      "metadata": {
+        "id": "-qFE_cKF5958"
+      },
+      "outputs": [],
+      "source": [
+        "# Note: Remove \"mmh\" when NNPOps in available on conda-forge (https://github.com/openmm/NNPOps/issues/26)\n",
+        "# Note: \"cudatoollit=11.2\" is need to override a pinning in conda-colab\n",
+        "!conda install -q -c conda-forge -c mmh \\\n",
+        "               openmm-torch nnpops torchani openmmtools \\\n",
+        "               cudatoolkit=11.2 \\\n",
+        "               &> /dev/null # Comment this line to see the installaion logs"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ZQwKdquUF5wo"
+      },
+      "source": [
+        "## Prepare a simulation system\n",
+        "\n",
+        "For simplicity, the alanine dipeptide system from [OpenMM-Tools](https://openmmtools.readthedocs.io/en/latest/) is used."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 1,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "zNng_d4mGKve",
+        "outputId": "218e0775-766b-4597-9d48-96e3d65df374"
+      },
+      "outputs": [
+        {
+          "name": "stderr",
+          "output_type": "stream",
+          "text": [
+            "Warning: importing 'simtk.openmm' is deprecated.  Import 'openmm' instead.\n"
+          ]
+        }
+      ],
+      "source": [
+        "import openmmtools\n",
+        "\n",
+        "# Get the system of alanine dipeptide\n",
+        "ala2 = openmmtools.testsystems.AlanineDipeptideVacuum(constraints=None)\n",
+        "\n",
+        "# Remove MM forces\n",
```

Comment:
What about refactoring the process of removing MM forces and creating/adding the ML potential into a small function folks can copy-and-paste into their own code?

For example, this code could be bundled together:
```python
def replace_mm_with_nnp(system, topology, NNP):
    """
    Create an instance of the specified machine learning potential and replace all MM forces with the ML potential.

    """
    # Remove MM forces
    while system.getNumForces() > 0:
        system.removeForce(0)

    # Assert no forces or constraints remain
    assert system.getNumConstraints() == 0 
    assert system.getNumForces() == 0

    # Get atomic numbers
    atomic_numbers = [ atom.element.atomic_number for atom in topology.atoms() ]

    # Create an instance of the model
    nnp = NNP(atomic_numbers)
   
    # Save the NNP to a file and load it with OpenMM-Torch
    # TODO: Use a temporary file instead
    pt.jit.script(nnp).save('model.pt')
    force = TorchForce('model.pt')

    # Add the NNP to the system
    system.addForce(force)
    assert ala2.system.getNumForces() == 1 
```

Reply:
As commented below, this a low-level tutorial. All the convenience functionality should be in OpenMM-ML.